docs/codeql/codeql-cli/about-codeql-packs.rst

```diff
@@ -0,0 +1,104 @@
+files and directories within the pack should be logically organized. For example, typically:
+
+- Queries are organized into directories for specific categories.
+- Queries for specific products, libraries, and frameworks are organized into
+  their own top-level directories.
+
+About ``qlpack.yml`` files
+--------------------------
+
+When executing query-related commands, CodeQL first looks in the package cache for CodeQL packs which have already been downloaded. If a suitable pack cannot be found, then CodeQL scans siblings of the installation directory (and their subdirectories) for ``qlpack.yml`` files. The metadata in the file tells
+CodeQL how to compile queries, what libraries the pack depends on, and where to
+find query suite definitions.
+
+The content of the CodeQL pack (queries and libraries used in CodeQL analysis) is
+included in the same directory as ``qlpack.yml``, or its subdirectories.
+
+The location of ``qlpack.yml`` defines the library path for the content
+of the CodeQL pack. That is, for all ``.ql`` and ``.qll`` files in the pack,
+CodeQL will resolve all import statements relative to the ``qlpack.yml`` at the
+pack's root.
+
+.. _codeqlpack-yml-properties:
+
+``qlpack.yml`` properties
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following properties are supported in ``qlpack.yml`` files.
+
+.. list-table::
+   :header-rows: 1
+   :widths: auto
+
+   * - Property
+     - Example
+     - Required
+     - Purpose
+   * - ``name``
+     - ``octo-org/security-queries``
+     - All packs
+     - The scope, where the CodeQL pack is published, and the name of the pack defined using alphanumeric characters, hyphens, and periods. It must be unique as CodeQL cannot differentiate between CodeQL packs with identical names. Name components cannot start or end with a hyphen. Additionally, a period is not allowed in pack names at all. Use the pack name to specify queries to run using ``database analyze`` and to define dependencies between QL packs (see examples below).
+   * - ``version``
+     - ``0.0.0``
+     - All packs
+     - A version number for this CodeQL pack. This must be a valid semantic version that meets the `SemVer v2.0.0 specification <https://semver.org/spec/v2.0.0.html>`__.
+   * - ``dependencies``
+     - ``codeql/javascript-all: 1.2.3``
+     - Optional
+     - The names of any CodeQL packs that this pack depends on, as a sequence. This gives the pack access to any libraries, database schema, and query suites defined in the dependency.
+   * - ``suites``
+     - ``octo-org-query-suites``
+     - Optional
+     - The path to a directory that contains the "well-known" query suites in the pack, defined relative to the pack directory. You can run "well-known" suites stored in this directory by specifying the pack name, without providing their full path. For more information about query suites, see ":doc:`Creating CodeQL query suites <creating-codeql-query-suites>`."
```

Comment:
@dbartol or @aeisenberg can you shed any light on what "well-known suites" are? Is this how we make the standard suites available so that you just call:

```
codeql database analyze /codeql-dbs/example-repo  javascript-code-scanning.qls ...
```

I think this needs a minor rewrite because currently it sounds as if "well-known suites" are a thing already, but I _think_ this property is how you create a "well-known suite". If this is the case, then I think "alias" or "short cut" might be clearer.

Is that right or is something else going on here?

Reply:
Well-known suites are a legacy (pre-packaging) way of making it easy to discover suites to run. We have retained this behaviour for qlpacks and suites that are outside of the package store (ie- those that have not been downloaded from a registry), but we have not extended this feature to working with downloaded packs (we may do this in the future, but it is lower priority since we don't expect users to take advantage of this very much).

Any qlpacks that are discoverable by the CLI (but that are outside of the package store) expose _well-known suites_ through the `suites` property in the qlpack.yml file. The `suites` property points to a directory. Every `qls` file in that directory is considered a _well-known suite_.

Reply:
🤦🏻 I should have looked at the link. I hadn't realized that this was such a well-established definition. It still seems as if the name chosen could be clearer, but this is obviously something to leave. I've just revised the definition so it's clearer.